model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -178,6 +266,82 @@
 DOWNSTREAM_REQUEST_TIMEOUT_SECONDS = 5 * 60  # 5 minutes
 
 
+# Hack to count prompt tokens
+tokenizer_cache: Dict[str, AutoTokenizer] = {}
```

Comment:
nit: would we want to make this an lru cache or something? idk how big the tokenizers can get

Reply:
~3MB per model. good call to add lru cache